### PR TITLE
Correlation matrix optimization

### PR DESCRIPTION
**Pull request by cshekhar17**

### CODE REVIEW

**Comment by kajanansangar** [comment]
@sourjyas you should try adding a transformer to convert categorical to numerical before passing to correlation_matrix_numerical, we can not ignore this in the config.yaml

File: src/test/anovos/data_analyzer/test_association_evaluator.py [hunk 1]
```diff
@@ -355,196 +357,65 @@ def test_variable_clustering(spark_session: SparkSession):
...
+        result_df3.where(F.col("attribute") == "fnlwgt")
         .toPandas()
-        .to_dict("list")["education"][0]
-        <= 0.3
+        .to_dict("list")["logfnl"][0]
+        <= 0.95
     )
     assert (
-        result_df3.where(F.col("attribute") == "capital-gain")
+        result_df3.where(F.col("attribute") == "hours-per-week")
```
Comment (kajanansangar) [inline src/test/anovos/data_analyzer/test_association_evaluator.py:411]: why did you change the column names here?

File: src/main/anovos/data_transformer/transformers.py [hunk 2]
```diff
@@ -423,6 +424,84 @@ def monotonic_binning(
     return odf
 
 
+def cat_to_num_transformer(
```
Comment (kajanansangar) [inline src/main/anovos/data_transformer/transformers.py:427]: this method should support the supervised method type too. We introduce two additional parameters such as method_type={"supervised" or "unsupervised"}
encoding="one-hot" or "label". We need to have the right error handling technique to avoid the wrong combination of inputs

**Review by kajanansangar — CHANGES_REQUESTED** [review]

File: config/configs_segmentation_unsupervised.yaml [hunk 3]
```diff
@@ -88,6 +88,13 @@ quality_checker:
       output_mode: replace
 
 
+cat_to_num_transformer:
+  list_of_cols: all
+  drop_cols: 'ifa'
+  method_type: label_encoding
+  label_col: income
```
[the remark this answers is not in the document]
Comment (sourjyasen07) [reply config/configs_segmentation_unsupervised.yaml]: But this has been changed already?

File: config/configs_segmentation_unsupervised.yaml [hunk 4]
```diff
@@ -88,6 +88,13 @@ quality_checker:
...
+  event_label: '>50K'
```
[the remark this answers is not in the document]
Comment (sourjyasen07) [reply config/configs_segmentation_unsupervised.yaml]: But this has been changed already?

File: src/main/anovos/data_transformer/transformers.py (hunk 2 above)
Comment (sourjyasen07) [reply src/main/anovos/data_transformer/transformers.py:427]: checked in the changes based on the discussion

File: src/test/anovos/data_analyzer/test_association_evaluator.py (hunk 1 above)
Comment (sourjyasen07) [reply src/test/anovos/data_analyzer/test_association_evaluator.py:411]: these test cases were not added by me, may be @ChandraShekhar2 ?

File: src/main/anovos/data_transformer/transformers.py [hunk 5]
```diff
@@ -423,6 +424,84 @@ def monotonic_binning(
...
+        drop_cols argument is not considered for analysis even if it is mentioned in list_of_cols. (Default value = "all")
+    drop_cols
+        List of columns to be dropped e.g., ["col1","col2"].
+        Alternatively, columns can be specified in a string format,
+        where different column names are separated by pipe delimiter “|” e.g., "col1|col2".
+        It is most useful when coupled with the “all” value of list_of_cols, when we need to consider all columns except
+        a few handful of them. (Default value = [])
+    label_col
+        Label/Target column (Default value = "label")
+    event_label
```
[the remark this answers is not in the document]
Comment (sourjyasen07) [reply src/main/anovos/data_transformer/transformers.py]: yes, makes sense.. changes reflected in the latest commit..

File: src/main/anovos/data_transformer/transformers.py [hunk 6]
```diff
@@ -423,6 +424,84 @@ def monotonic_binning(
     return odf
 
 
+def cat_to_num_transformer(
+    spark,
+    idf,
+    list_of_cols="all",
+    drop_cols=[],
+    label_col="label",
```
[the remark this answers is not in the document]
Comment (sourjyasen07) [reply src/main/anovos/data_transformer/transformers.py]: yes, makes sense.. changes reflected in the latest commit..

File: config/configs_segmentation_unsupervised.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sourjyasen07) [reply config/configs_segmentation_unsupervised.yaml]: changes reflected in the latest commit..

File: config/configs_segmentation_unsupervised.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sourjyasen07) [reply config/configs_segmentation_unsupervised.yaml]: changes reflected in the latest commit..

**Review by kajanansangar — APPROVED** [review]
